webapp/channels/src/components/advanced_text_editor/advanced_text_editor.test.tsx

```diff
@@ -165,4 +193,186 @@ describe('components/avanced_text_editor/advanced_text_editor', () => {
             expect(textbox).not.toHaveFocus();
         });
     });
+
+    it('should set the textbox value to an existing draft on mount and when changing channels', () => {
+        const {rerender} = renderWithContext(
+            <AdvancedTextEditor
+                {...baseProps}
+            />,
+            mergeObjects(initialState, {
+                storage: {
+                    storage: {
+                        [StoragePrefixes.DRAFT + channelId]: {
+                            value: TestHelper.getPostDraftMock({
+                                message: 'original draft',
+                            }),
+                        },
+                        [StoragePrefixes.DRAFT + otherChannelId]: {
+                            value: TestHelper.getPostDraftMock({
+                                message: 'a different draft',
+                            }),
+                        },
+                    },
+                },
+            }),
+        );
+
+        expect(screen.getByLabelText('write to test channel')).toHaveValue('original draft');
+
+        rerender(
+            <AdvancedTextEditor
+                {...baseProps}
+                channelId={otherChannelId}
+            />,
+        );
+        expect(screen.getByLabelText('write to other channel')).toHaveValue('a different draft');
+    });
+
+    it('should save a new draft when changing channels', () => {
+        const {rerender} = renderWithContext(
+            <AdvancedTextEditor
+                {...baseProps}
+            />,
+            initialState,
+        );
+
+        userEvent.type(screen.getByLabelText('write to test channel'), 'some text');
+
+        expect(updateDraft).not.toHaveBeenCalled();
+
+        rerender(
+            <AdvancedTextEditor
+                {...baseProps}
+                channelId={otherChannelId}
+            />,
+        );
+
+        expect(updateDraft).toHaveBeenCalled();
+        expect((updateDraft as unknown as jest.Mock<typeof updateDraft>).mock.calls[0][1]).toMatchObject({
```

Comment:
To avoid having to do all these castings, you can use `jest.mocked` as used in `webapp/channels/src/components/suggestion/emoticon_provider.test.tsx`.

Reply:
Good suggestion. I don't think I've seen that before, but if I have, I forgot about it. I'll change that